source/shared/core_stmt.cpp

```diff
@@ -412,387 +364,75 @@ void core_sqlsrv_bind_param( _Inout_ sqlsrv_stmt* stmt, _In_ SQLUSMALLINT param_
-            break;
-    }
-
-    if( zval_was_null ){
-        ind_ptr = SQL_NULL_DATA;
-    }
-
-    core::SQLBindParameter( stmt, param_num + 1, direction,
-        c_type, sql_type, column_size, decimal_digits, buffer, buffer_len, &ind_ptr );
 
+        // Get all necessary values to prepare for SQLBindParameter
+        param_ptr->process_param(stmt, param_z);
+        param_ptr->bind_param(stmt);
 
-    // When calling SQLDescribeParam() on a parameter targeting a Datetime column, the return values for ParameterType, ColumnSize and DecimalDigits are SQL_TYPE_TIMESTAMP, 23, and 3 respectively.
-    // For a parameter targeting a SmallDatetime column, the return values are SQL_TYPE_TIMESTAMP, 16, and 0. Inputting these values into SQLBindParameter() results in Operand type clash error.
-    // This is because SQL_TYPE_TIMESTAMP corresponds to Datetime2 by default, and conversion of Datetime2 to Datetime and conversion of Datetime2 to SmallDatatime is not allowed with encrypted columns.
-    // To fix the conversion problem, set the SQL_CA_SS_SERVER_TYPE field of the parameter to SQL_SS_TYPE_DATETIME and SQL_SS_TYPE_SMALLDATETIME respectively for a Datetime and Smalldatetime column.
-    // Note this must be called after SQLBindParameter() or SQLSetDescField() may fail. 
-    // TODO: how to correctly distinguish datetime from datetime2(3)? Both have the same decimal_digits and column_size
-    if (stmt->conn->ce_option.enabled && sql_type == SQL_TYPE_TIMESTAMP) {
-        if (decimal_digits == 3) {
-            core::SQLSetDescField(stmt, param_num + 1, SQL_CA_SS_SERVER_TYPE, (SQLPOINTER)SQL_SS_TYPE_DATETIME, SQL_IS_INTEGER);
-        } else if (decimal_digits == 0 && column_size == 16) {
-            core::SQLSetDescField(stmt, param_num + 1, SQL_CA_SS_SERVER_TYPE, (SQLPOINTER)SQL_SS_TYPE_SMALLDATETIME, SQL_IS_INTEGER);
+        // When calling SQLDescribeParam() on a parameter targeting a Datetime column, the return values for ParameterType, ColumnSize and DecimalDigits are SQL_TYPE_TIMESTAMP, 23, and 3 respectively.
+        // For a parameter targeting a SmallDatetime column, the return values are SQL_TYPE_TIMESTAMP, 16, and 0. Inputting these values into SQLBindParameter() results in Operand type clash error.
+        // This is because SQL_TYPE_TIMESTAMP corresponds to Datetime2 by default, and conversion of Datetime2 to Datetime and conversion of Datetime2 to SmallDatatime is not allowed with encrypted columns.
+        // To fix the conversion problem, set the SQL_CA_SS_SERVER_TYPE field of the parameter to SQL_SS_TYPE_DATETIME and SQL_SS_TYPE_SMALLDATETIME respectively for a Datetime and Smalldatetime column.
+        // Note this must be called after SQLBindParameter() or SQLSetDescField() may fail. 
+        // TODO: how to correctly distinguish datetime from datetime2(3)? Both have the same decimal_digits and column_size
```

Comment:
Is this still a valid TODO, or are we able to address?

Reply:
Yes, it's still in the backlog